doc/communicationDiagram.md

```diff
@@ -2,76 +2,98 @@
 
-## User Events
+## List of User Events
 
-This is a list of user events in TcpServer, with a numeric id to indetify them in the Communication Diagram.
+The following is the list of user events in **TcpServer**:
 
 | id | User Event |
 |:--:|:----------:|
-| 1 | CmdWrite |
-| 2 | CmdRead  |
-| 3 | EvtWrite |
-| 4 | EvtRead  |
-| 5 | GenWrite |
-| 6 | GenRead  |
-
-The following table shows the user events with their publisher and subscriber.
-
-| User Event | id | Publisher | Subscriber |
-|:--------:|:----:|----------:|-----------:|
-| CmdRead  | 2 | TcpServerCmd:processCmd.vi  | ComponentSimulator.vi     |
-| CmdWrite | 1 | TcpServer:sendCmdStatus.vi  | TcpServerCmd.runServer.vi |
-| EvtRead  | 4 | TcpServerCmd:processEvt.vi  | ComponentSimulator.vi     |
-| EvtWrite | 3 | TcpServer.sendEvt.vi         | TcpServerCmd.runServer.vi |
-| GenRead  | 6 | TcpServer:getClientStatus.vi | ComponentSimulator.vi     |
-| GenWrite | 5 |                              | TcpServerCmd:runServer.vi |
-|          |   | TcpServerCmd.runServer.vi    |                           |
-
-## Top Level User Events
-
-We will define a group of user events closest to the User and we will name them `Top Level User Events`.
-These user events are what the User needs to know and use.
-They are defined in the following table.
+| 1  | CmdRead    |
+| 2  | EvtRead    |
+| 3  | GenRead    |
+| 4  | CmdWrite   |
+| 5  | EvtWrite   |
+| 6  | GenWrite   |
+
+- `CmdRead` is used to send the `tcpPacket` to the Component if the incoming command is legal.
+- `EvtRead` is used to send the `tcpPacket` to the Component if the incoming event is legal.
+- `GenRead` is used to send the `tcpPacket` to the Component if the incoming message is unknown.
+- `CmdWrite` is used to send the ACK to the TCP client with the command result (success or fail).
+- `EvtWrite` is used to send an event to the TCP client from the Component.
+- `GenWrite`  not used. Ticket [DM-32272](https://jira.lsstcorp.org/browse/DM-32272) was created to evaluate this user event is needed or not.
+
+The following table shows the user events with their publishers and subscribers:
+
+| User Event | Publisher | Subscriber |
+|:--------:|:----------:|:-----------:|
+| CmdRead  | TcpServerCmd.processCmd.vi   | ComponentSimulator.vi     |
+| EvtRead  | TcpServerCmd.processEvt.vi   | ComponentSimulator.vi     |
+| GenRead  | TcpServer.getClientStatus.vi | ComponentSimulator.vi     |
+| GenRead  | TcpServerCmd.runServer.vi    | ComponentSimulator.vi     |
+| CmdWrite | TcpServer.sendCmdStatus.vi   | TcpServerCmd.runServer.vi |
+| EvtWrite | TcpServer.sendEvt.vi         | TcpServerCmd.runServer.vi |
+| GenWrite |              N/A             | TcpServerCmd.runServer.vi |
+
+This table shows the vi that publishes an user event and which vi subscribes to that user event.
+
+**TcpServerCmd** is the class that handles commands and events, and **TcpServerTel** is the class that handles telemetry messages.
```

Comment:
**TcpServerCmd** is the class that handles commands and events.
**TcpServerTel** is the class that handles telemetry messages.

Reply:
Done.